src/registrar/models/domain.py

```diff
@@ -1026,10 +1033,49 @@ def _remove_client_hold(self):
             # if registry error occurs, log the error, and raise it as well
             logger.error(f"registry error removing client hold: {err}")
             raise (err)
-
+        
     def _delete_domain(self):
         """This domain should be deleted from the registry
         may raises RegistryError, should be caught or handled correctly by caller"""
+
+        logger.info("Deleting subdomains for %s", self.name)
+        # check if any subdomains are in use by another domain
+        hosts = Host.objects.filter(name__regex=r".+{}".format(self.name))
+        for host in hosts:
+            if host.domain != self:
+                logger.error("Unable to delete host: %s is in use by another domain: %s", host.name, host.domain)
+                raise RegistryError(
+                    code=ErrorCode.OBJECT_ASSOCIATION_PROHIBITS_OPERATION,
+                    note=f"Host {host.name} is in use by {host.domain}",
+                )
+        
```

Comment:
That's a good point that I had considered, but in general there's not a great way to handle it. I haven't found a method in EPP that lets us list object based on their relationships, so we have to rely on our database foreign key relationships being an accurate-enough reflection. That said, it would be a relatively simple task to add more error handling if need be.